kg_microbe/transform_utils/madin_etal/madin_etal.py

```diff
@@ -297,78 +355,148 @@ def run(self, data_file: Union[Optional[Path], Optional[str]] = None, show_statu
-                                chebi_result_for_tax_id[OBJECT_LABEL_COLUMN]
-                                == chebi_result_for_tax_id[SUBJECT_LABEL_COLUMN]
-                            )
-                            exact_match_chebi_df = chebi_result_for_tax_id[exact_condition_chebi]
-                            if not exact_match_chebi_df.empty:
-                                chebi_result_for_tax_id = exact_match_chebi_df
-                            for row in chebi_result_for_tax_id.iterrows():
+                        # First try to find mappings in METPO
+                        carbon_substrates_not_in_metpo = []
+                        for substrate in carbon_substrates:
+                            metpo_mapping = self.madin_metpo_mappings.get(substrate.strip(), None)
+                            # print(f"Substrate: {substrate}, METPO mapping: {metpo_mapping}")
+                            if metpo_mapping:
+                                # create carbon substrate node and edge to tax_id
+                                # use `biolink equivalent` for 'category' and
+                                # for 'predicate', `biolink equivalent` from properties
+                                category = metpo_mapping.get(
+                                    "biolink_equivalent", CARBON_SUBSTRATE_CATEGORY
+                                )
+                                predicate_biolink = metpo_mapping.get(
+                                    "predicate_biolink_equivalent", ""
+                                )
+                                # fallback: if no biolink equivalent use `biolink:consumes`
+                                if predicate_biolink:
+                                    predicate = predicate_biolink
+                                else:
+                                    predicate = "biolink:consumes"
                                 carbon_substrate_nodes.append(
                                     [
-                                        row[1].object_id,
-                                        CARBON_SUBSTRATE_CATEGORY,
-                                        row[1].object_label,
+                                        metpo_mapping["curie"],
+                                        category,
+                                        metpo_mapping["label"],
                                     ]
                                 )
                                 tax_carbon_substrate_edge.append(
                                     [
                                         tax_id,
-                                        NCBI_TO_CARBON_SUBSTRATE_EDGE,
-                                        row[1].object_id,
-                                        BIOLOGICAL_PROCESS,
+                                        predicate,
+                                        metpo_mapping["curie"],
+                                        TROPHICALLY_INTERACTS_WITH,
```

Comment:
@sujaypatil96 also curious about TROPHICALLY_INTERACTS_WITH -- is that in the relation field?

Reply:
```
TROPHICALLY_INTERACTS_WITH = (
    "RO:0002438"  # [org_name -> 'trophically interacts with' -> carbon_substrate]
)
```

Yup, that value appears in the relation column.